regression-test/suites/nereids_p0/expression/fold_constant/fold_constant_string_arithmatic.groovy

```diff
@@ -16,763 +16,646 @@
+    testFoldConst("select lpad(cast('北京' as string), 5, cast('杭州' as string))")
+    
+    // ltrim
+    testFoldConst("select ltrim(' 11111', 11)")
+    testFoldConst("select ltrim('11111 ', 11)")
+    testFoldConst("select ltrim('   ab d')")
+    testFoldConst("select ltrim(cast(' 11111' as string), cast(11 as string))")
+    testFoldConst("select ltrim(cast('11111 ' as string), cast(11 as string))")
+    testFoldConst("select ltrim(cast('   ab d' as string))")
+    testFoldConst("select ltrim(cast('Hello' as string))")
+    testFoldConst("select ltrim(cast('  Hello World  ' as string))")
+    testFoldConst("select ltrim('Hello')")
+    testFoldConst("select ltrim('  Hello World  ')")
+    testFoldConst("select ltrim('  上海天津北京杭州  ')")
+    
+    // md5
+    testFoldConst("select md5(cast('Hello World' as string))")
+    testFoldConst("select md5('Hello World')")
+    testFoldConst("select md5('こんにちは')")
+    testFoldConst("select md5sum('Hello World')")
+    testFoldConst("select md5sum('こんにちは')")
+    testFoldConst("select md5sum('===*+-')")
+
+    // money_format
     testFoldConst("select money_format(1123.4)")
+    testFoldConst("select money_format(1123.456)")
+    testFoldConst("select money_format(17014116)")
     testFoldConst("select money_format(truncate(1000,10))")
-
-    testFoldConst("select null_or_empty(null)")
-    testFoldConst("select null_or_empty(\"\")")
-    testFoldConst("select null_or_empty(\"a\")")
-
-    testFoldConst("select not_null_or_empty(null)")
+    testFoldConst("select money_format(-1123.4)")
+    testFoldConst("select money_format(-1123.456)")
+    testFoldConst("select money_format(-17014116)")
+    testFoldConst("select money_format(-truncate(1000,10))")
+    
+    // not_null_or_empty
     testFoldConst("select not_null_or_empty(\"\")")
     testFoldConst("select not_null_or_empty(\"a\")")
-
-    testFoldConst("SELECT repeat(\"a\", 3)")
-    testFoldConst("SELECT repeat(\"a\", -1)")
-    testFoldConst("SELECT repeat(\"a\", 0)")
-    testFoldConst("SELECT repeat(\"a\",null)")
-    testFoldConst("SELECT repeat(null,1)")
-
+    testFoldConst("select not_null_or_empty(cast('a' as string))")
+    testFoldConst("select not_null_or_empty(cast('' as string))")
+    testFoldConst("select not_null_or_empty(null)")
+    testFoldConst("select not_null_or_empty(NULL)")
+    testFoldConst("select not_null_or_empty('\b')")
+    testFoldConst("select not_null_or_empty(' \b')")
+    
+    // null_or_empty
+    testFoldConst("select null_or_empty(\"\")")
+    testFoldConst("select null_or_empty(\"a\")")
+    testFoldConst("select null_or_empty(cast('a' as string))")
+    testFoldConst("select null_or_empty(cast('' as string))")
+    testFoldConst("select null_or_empty(null)")
+    testFoldConst("select null_or_empty(NULL)")
+    testFoldConst("select null_or_empty('\b')")
+    testFoldConst("select null_or_empty(' \b')")
+    
+    // overlay
+    testFoldConst("select overlay('abcdef', '123', 3, 2)")
```

Comment:
the `overlay` syntax is `VARCHAR Overlay (VARCHAR str, INT pos, INT len, VARCHAR newstr)`, the test case set the run parameter for it